tools/topology/topology2/platform/intel/dmic-generic.conf

```diff
@@ -391,13 +404,26 @@ IncludeByKey.PASSTHROUGH {
 					type		dai_out
 					stream_name	$DMIC0_NAME
 					node_type $DMIC_LINK_INPUT_CLASS
-					num_input_audio_formats	2
+					num_input_audio_formats	4
 					Object.Base.input_audio_format [
+						{
+							in_rate			$DMIC0_RATE
+							in_bit_depth		16
+							in_valid_bit_depth	16
+						}
 						{
 							in_rate			$DMIC0_RATE
 							in_bit_depth		32
 							in_valid_bit_depth	32
 						}
+						{
+							in_rate			$DMIC0_RATE
+							in_channels		4
+							in_bit_depth		16
+							in_valid_bit_depth	16
+							in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+							in_ch_map	$CHANNEL_MAP_3_POINT_1
+						}
```

Comment:
should we also add support for 24 bits in 32-bit container if this is supported in external NHLTs?

Reply:
The 24 in 32 format is the same as 32 in 32, the kernel is treating both as S32_LE

Reply:
erm, what?

we clearly have instances where we list both formats:

````
Class.Pipeline."io-gateway" {

	<include/pipelines/pipeline-common.conf>

	attributes {
		!constructor [
			"index"
		]

		#
		# io-gateway objects instantiated within the same alsaconf node must have
		# unique instance attribute
		#
		unique	"instance"
	}

	Object.Widget {
		dai-copier."1" {
			type dai_in
			num_input_audio_formats 3
			Object.Base.input_audio_format [
				{
					in_bit_depth            32
					in_valid_bit_depth      24
				}
				{
					in_bit_depth		32
					in_valid_bit_depth	32
				}	
````

Reply:
@plbossart , what I meant is that for the DMIC blob bit depth the 24/32 and 32/32 is the same. It is basically S32_LE msbits=24 (just like HDA).
For both description the blob itself is identical.
I guess it should not hurt, but it is kind of redundant?

Reply:
I agree, why not add it too, then it should be capable to support everything future may bring to it.